cli_tools/gce_ovf_export/domain/params.go

```diff
@@ -0,0 +1,132 @@
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package ovfexportdomain
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/daisycommon"
+)
+
+const (
+	// InstanceNameFlagKey is key for instance name CLI flag
+	InstanceNameFlagKey = "instance-name"
+
+	// MachineImageNameFlagKey is key for machine image name CLI flag
+	MachineImageNameFlagKey = "machine-image-name"
+
+	// ClientIDFlagKey is key for client ID CLI flag
+	ClientIDFlagKey = "client-id"
+
+	// DestinationURIFlagKey is key for OVF/OVA GCS path CLI flag
+	DestinationURIFlagKey = "destination-uri"
+
+	// ReleaseTrackFlagKey is key for release track flag
+	ReleaseTrackFlagKey = "release-track"
+
+	// OvfFormatFlagKey is key for OVF format flag
+	OvfFormatFlagKey = "ovf-format"
+
+	mainWorkflowDir = "daisy_workflows/"
+)
+
+// OVFExportParams holds flags for OVF export
+type OVFExportParams struct {
+
+	// Instance export specific flags
+	InstanceName string
+
+	// Machine image specific flags
+	MachineImageName string
+
+	// Common flags
+	ClientID             string
+	DestinationURI       string
+	OvfFormat            string
+	DiskExportFormat     string
+	OsID                 string
+	Network              string
+	Subnet               string
+	NoExternalIP         bool
+	Zone                 string
+	Timeout              time.Duration
+	Project              *string
+	ScratchBucketGcsPath string
+	Oauth                string
+	Ce                   string
+	GcsLogsDisabled      bool
+	CloudLogsDisabled    bool
+	StdoutLogsDisabled   bool
+	ReleaseTrack         string
+	BuildID              string
+
+	// Non-flags
+	WorkflowDir string
+	Region      string
+	Started     time.Time
+}
+
+func (params *OVFExportParams) String() string {
+	return fmt.Sprintf("%#v", params)
+}
+
+// IsInstanceExport returns true if export represented by these params is
+// instance export. False otherwise.
+func (params *OVFExportParams) IsInstanceExport() bool {
+	return params.InstanceName != ""
+}
+
+// IsMachineImageExport returns true if export represented by these params is
+// a machine image export. False otherwise.
+func (params *OVFExportParams) IsMachineImageExport() bool {
+	return !params.IsInstanceExport()
+}
+
+//TODO: consolidate with ovf_importer.toWorkingDir
+func toWorkingDir(currentDir, workflowDir string) string {
+	wd, err := filepath.Abs(filepath.Dir(currentDir))
+	if err == nil {
+		return path.Join(wd, workflowDir)
+	}
+	return workflowDir
+}
+
+// InitWorkflowPath initializes workflow path field
+func (params *OVFExportParams) InitWorkflowPath() {
+	currentExecutablePath := filepath.Dir(os.Args[0])
+	params.WorkflowDir = toWorkingDir(currentExecutablePath, mainWorkflowDir)
```

Comment:
This only builds a path without checking it exists. I don't think we actually check whether working dir exists before passing it to Daisy. Do you have examples if we actually do that?

Reply:
I've found it useful when developing, especially since paths might be different in local development, test, and production. That said, it's definitely just a suggestion :)  Feel free to resolve if you don't think it'll be helpful.

Reply:
Do we do that already in any of the other wrappers, so I don't have to re-invent the wheel?

Reply:
In terms of validation, the closest I found is the link below. I checked args.go in gce_vm_image_import and didn't see validations on the workflow directory.

- https://github.com/GoogleCloudPlatform/compute-image-tools/blob/5dd7394ac6f88c81a7ce13121b31ff62c987793a/daisy/sources.go#L171

If you find the validation useful, here are some things that you could leverage. The first is an existing predicate, and the second is a package that contains assertions.

- https://github.com/GoogleCloudPlatform/compute-image-tools/blob/aa1cca8adc9530bf24a69e1b290a8d7dcf74127b/cli_tools/common/utils/files/files.go#L22
- https://github.com/GoogleCloudPlatform/compute-image-tools/blob/f5a91c65376c26d14c93854a42d408fc2da41a81/cli_tools/common/assert/assert.go#L15